projects/aca-playwright-shared/src/api/sites-api.ts

```diff
@@ -63,20 +63,51 @@ export class SitesApi {
     }
   }
 
-      /**
+  /**
    * Delete multiple sites/libraries.
    * @param siteIds The list of the site/library IDs to delete.
    * @param permanent Delete permanently, without moving to the trashcan? (default: true)
    */
-      async deleteSites(siteIds: string[], permanent: boolean = true) {
-        try {
-          if (siteIds && siteIds.length > 0) {
-            for (const siteId of siteIds) {
-              await this.apiService.sites.deleteSite(siteId, { permanent });
-            }
-          }
-        } catch (error) {
-          logger.error(`${this.constructor.name} ${this.deleteSites.name}`, error);
+  async deleteSites(siteIds: string[], permanent: boolean = true) {
+    try {
+      if (siteIds && siteIds.length > 0) {
+        for (const siteId of siteIds) {
+          await this.apiService.sites.deleteSite(siteId, { permanent });
         }
       }
+    } catch (error) {
+      logger.error(`${this.constructor.name} ${this.deleteSites.name}`, error);
+    }
+  }
+
+  async updateSiteMember(siteId: string, userId: string, role: string) {
+    const siteRole = {
+      role: role
+    } as SiteMembershipBodyUpdate;
+
+    try {
+      return await this.apiService.sites.updateSiteMembership(siteId, userId, siteRole);
+    } catch (error) {
+      logger.error(`SitesApi updateSiteMember : catch : `, error);
+      return null;
```

Comment:
You can probably return empty object instead of null not to add it to the return type

Reply:
done